draft-packages/title-block-zen/KaizenDraft/TitleBlockZen/MobileActions.tsx

```diff
@@ -18,44 +20,31 @@ import styles from "./MobileActions.module.scss"
 
 const buttonIsLink: (action: TitleBlockButtonProps) => boolean = action =>
   "href" in action
+
 const buttonIsAction: (action: TitleBlockButtonProps) => boolean = action =>
   !("href" in action) && "onClick" in action
 
-const renderPrimaryLinks = (
+const renderPrimaryActionDrawerContent = (
   primaryAction: PrimaryActionProps
 ): JSX.Element[] | null | undefined => {
   if (!primaryAction) return null
   if (isMenuGroupNotButton(primaryAction)) {
-    return primaryAction.menuItems
-      .filter(item => typeof item.action === "string")
-      .map((item, idx) => (
-        <MenuItem
+    return primaryAction.menuItems.map((item, idx) => {
+      const itemType =
```

Comment:
See, I _like_ this stuff, but people keep telling me to make this sort of stuff a function. I think in a design system context you probably wanna be explicit about what this is trying to find out?

Reply:
Fair, I can probably break it into smaller pieces to try to increase legibility